liferay-echo/liferay-portal
# LPD-27086 Avoid looking up friendlyURLEntries from the exporting site

Motivation
=======
Staging or Export/Import can fail with a DuplicateFriendlyURLException if the Exporting site is in the same instance as the Importing site and the Import is happening for the second time.
https://liferay.atlassian.net/browse/LPD-27086

Proposed Solution
========
`FriendlyURLEntryStagedModelRepository#_getLocalizationMap` can get different `friendlyURLEntry`s based on whether its called from `addStagedModel` or `updateStagedModel` because `FriendlyURLEntryStagedModelDataHandler#doImportStagedModel` does not forward the same `importedFriendlyURLEntry` object to them.
We should to forward the same `importedFriendlyURLEntry` so that `FriendlyURLEntryStagedModelRepository` will always have access to the importing site's data.


Steps to Verify
========
1. Create a blank content page called “page1” and publish it
2. Create a blank content page called “page2” and publish it
3. Activate local staging
4. Configure “page1” and set its friendlyUrl to “/friendly”, save changes
5. Run a basic publication
6. Configure “page1” and change its friendlyUrl to “/one”, save changes
7. Without leaving the configuration screen of “page1”, from the friendlyUrl history forget the “/friendly” url
8. Configure “page2” and change its friendlyUrl to “/friendly”, save changes
9. In advanced publication, publish only “page2”
10. In advanced publication publish “page2” again

**Expected behavior:** Publication process ends without failures
**Actual behavior:** Publication fails, in the log you get the following exception:


PR TITLE & DESCRIPTION:
